### PR TITLE
Improve Simple Detections Testing - Also add ClassifiedAnalysis type

### PR DESCRIPTION
**Pull request by dekatzenel**
### Background

Ensure testing of Simple Detections functionality within PAT prior to public release
relates to: https://app.asana.com/0/1202324455056256/1204324416848382/f
relates to: https://app.asana.com/0/1202324455056256/1204630551936556/f

### Changes

* Add schema testing
* Add files to existing main tests
* Convert `(file_name, dir_name, spec)` tuple to new `ClassifiedAnalysis` class

### Testing

* Existing tests pass
* Added new tests

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.22.0
+0.22.1
```
Comment (lindsey-w) [inline VERSION:1]: we shouldn't release this yet

File: panther_analysis_tool/analysis_utils.py [hunk 2]
```diff
@@ -40,21 +40,31 @@
 )
 
 
+class ClassifiedAnalysis:
```
Comment (lindsey-w) [inline panther_analysis_tool/analysis_utils.py:43]: ❤️

File: tests/unit/panther_analysis_tool/test_schemas.py [hunk 3]
```diff
@@ -167,3 +167,156 @@ def test_invalid_props(self):
...
         with self.assertRaises(jsonschema.exceptions.ValidationError):
             self.call_validate(case)
+
+    def test_valid_scalar_match_any_type_no_value(self):
+        case = self.get_test_case()
+        case['Detection'] = [{"Key": "event_type",
+                              "Condition": "Exists"},
+                             {"DeepKey": ["details", "new_value"],
+                              "Condition": "IsNotNull"}]
+        RULE_SCHEMA.validate(case)
```
Comment (lindsey-w) [inline tests/unit/panther_analysis_tool/test_schemas.py:177]: I guess we could move this into `call_validate` as well

**Review by lindsey-w — DISMISSED** [review]
✨

**Review by lindsey-w — DISMISSED** [review]

File: panther_analysis_tool/analysis_utils.py (hunk 2 above)
Comment (maxrichie5) [inline panther_analysis_tool/analysis_utils.py:43]: Where did the name `Classified` come from?

**Review by maxrichie5 — APPROVED** [review]
Thanks!!!

File: panther_analysis_tool/analysis_utils.py (hunk 2 above)
Comment (dekatzenel) [reply panther_analysis_tool/analysis_utils.py:43]: It's derived from the name of the function that generates this type, `classify_analysis`. Happy to change it if there's something better

File: tests/fixtures/simple-detections/valid/AWS.EC2.Traffic.Mirroring.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# rights to access the Panther SaaS, are governed by the Panther Enterprise Subscription Agreement.
+
+AnalysisType: rule
+DisplayName: AWS EC2 Traffic Mirroring
+Severity: Medium
+Enabled: true
+LogTypes:
+  - AWS.CloudTrail
+RuleID: AWS.EC2.Traffic.Mirroring
+Detection:
```
[the remark this answers is not in the document]
Comment (dekatzenel) [reply tests/fixtures/simple-detections/valid/AWS.EC2.Traffic.Mirroring.yml:569]: resolved: [098f758](https://github.com/panther-labs/panther_analysis_tool/pull/289/commits/098f758691ee8dfbecc1c179d9684c224e693595)

[the remark this answers is not in the document]
Comment (dekatzenel) [reply tests/fixtures/simple-detections/valid/AWS.EC2.Traffic.Mirroring.yml:569]: To answer your question, it's a nice to have, but not needed

File: panther_analysis_tool/analysis_utils.py (hunk 2 above)
Comment (maxrichie5) [reply panther_analysis_tool/analysis_utils.py:43]: Just wondering. I have no better suggestion. But it doesn't tell me a whole lot about what the class does, if that helps at all

**Review by maxrichie5 — APPROVED** [review]

**Pull request merged**